# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Tenuring.h
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/src/gc/Tenuring.h
--- a/js/src/gc/Tenuring.h	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/src/gc/Tenuring.h	Fri Aug 04 14:06:49 2023 +0000
@@ -20,6 +20,10 @@
 class Nursery;
 class PlainObject;
 
+namespace wasm {
+class AnyRef;
+}  // namespace wasm
+
 namespace gc {
 
 class RelocationOverlay;
@@ -121,6 +125,7 @@
   size_t getTenuredCells() const;
 
   void traverse(JS::Value* thingp);
+  void traverse(wasm::AnyRef* thingp);
 
   // The store buffers need to be able to call these directly.
   void traceObject(JSObject* obj);